File: mission/elementary/digits-multiplication.js
"use strict";

function digitsMultip(data) {
    let dataString = data.toString();
    let result = 1;
    for (let i = 0; i < dataString.length; i++) {
        let buffer = Number(dataString[i]);
        result = buffer === 0 ? result : result * buffer;
    }
    return result;
}

var assert = require('assert');

if (!global.is_checking) {
    assert.equal(digitsMultip(123405), 120, "1st");
    assert.equal(digitsMultip(999), 729, "2nd");
    assert.equal(digitsMultip(1000), 1, "3rd");
    assert.equal(digitsMultip(1111), 1, "4th");
    console.log("Coding complete? Click 'Check' to review your tests and earn cool rewards!");
}

File: mission/python/elementary/digits-multiplication.py
def checkio(number):
    result = 1
    numbers = [int(num) for num in str(number) if num != '0']
    for digit in numbers:
        result = result * digit
    return result


if __name__ == '__main__':
    assert checkio(123405) == 120
    assert checkio(999) == 729
    assert checkio(1000) == 1
    assert checkio(1111) == 1
    print("Coding complete? Click 'Check' to review your tests and earn cool rewards!")
